hack/mock-unittest-data/gomod.py

```diff
@@ -0,0 +1,175 @@
+#!/usr/bin/env python3
+import json
+import subprocess
+from contextlib import contextmanager
+from functools import partial
+from itertools import chain
+from os import environ
+from pathlib import Path
+from shutil import copyfile
+from shutil import rmtree as delete_directory_with_contents
+from sys import argv
+from tempfile import gettempdir
+
+from yarn import print_banner
+
+
+def mkdir_p(dirname):
+    return Path(dirname).mkdir(exist_ok=True, parents=True)
+
+
+def generate_auxiliary_dirs(mocked_data_dir):
+    mkdir_p(f"{mocked_data_dir}/non-vendored")
+    mkdir_p(f"{mocked_data_dir}/vendored")
+    mkdir_p(f"{mocked_data_dir}/workspaces")
+    mocked_data_dir_abspath = Path(f"{mocked_data_dir}").absolute()
+
+    tmpdir = Path(gettempdir())
+
+    return mocked_data_dir_abspath, tmpdir
+
+
+@contextmanager
+def pandemonium_dir(pandemonium_root):
+    pandemo_dir = pandemonium_root / "gomod-pandemonium"
+    try:
+        yield pandemo_dir
+    finally:
+        delete_directory_with_contents(pandemo_dir)
+
+
+def inject_cwd(run, pandemo_dir):
+    _, _, fpart = run.__reduce__()
+    f, args, kwargs, _dct = fpart
+    kwargs |= {"cwd": pandemo_dir}
+    run.__setstate__((f, args, kwargs, _dct))
+
+
+def _c(command_string):
+    return command_string.split()
+
+
+def run_with_redirection(run, where, what, mocked_data_dir, mocked_data_dir_abspath):
```

Comment:
It took me half a minute to connect the name (redirection) with what you meant, likely it's just me.
nitpick: how about `write_to_file` instead?